sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala

```diff
@@ -44,14 +44,17 @@ object SQLExecution extends Logging {
 
   private def nextExecutionId: Long = _nextExecutionId.getAndIncrement
 
-  private val executionIdToQueryExecution = new ConcurrentHashMap[Long, QueryExecution]()
+  private[sql] val executionIdToQueryExecution = new ConcurrentHashMap[Long, QueryExecution]()
```

Comment:
Hm I like the current implementation/separation but doing this in `SQLExecution` might have some corner cases, e.g., `df.rdd.collect()` won't be cancelled. One way is to explicitly document that the cancellations only work with SQL/DataFrame API.

Reply:
BTW, does this work streaming queries too? I am fine with doing it in a followup but would like to make sure Spark Connect and Classic versions behave the same.

Reply:
Thank for the comment - I'll add it to the documentation.
Streaming - not sure, I'll check.